### ISSUE TITLE
No Settings or Scripts to block under Firefox

### ISSUE THREAD

**Issue by WUUUGI** [andryou/scriptsafe#313]
ScriptSafe does not to work for me. On every website there is no script in the dropdown menu that I could block and the options menu is empty (screenshot related).
![scriptsafe_bug](https://user-images.githubusercontent.com/24371969/34255154-388f5bb6-e647-11e7-9eb3-603e4c415557.png)

I am using Firefox Version 57.0.2 (64-bit) for Arch Linux under a newly created profile with uBlock Origin, HTTPS Everywhere, Decentraleyes, Cookie AutoDelete and Smart Referer.

**Comment by ghost** [comment]
I get the same issue. There is an error related to local storage in the console as well which is shown below this paragraph. It might be because of my extremely tweaked Firefox configuration which does not have storage related issues my other add-on - UBlock. Using `Firefox 57.0.2 (win64)` by the way.

```
SecurityError: The operation is insecure.
scriptsafe.js:610
  optionExists - moz-extension://89055bcf-d49f-43f9-bb5b-b0de5e5a2481/js/scriptsafe.js:610:10
  checkLocalStorage - moz-extension://89055bcf-d49f-43f9-bb5b-b0de5e5a2481/js/scriptsafe.js:624:7
  <anonymous> - moz-extension://89055bcf-d49f-43f9-bb5b-b0de5e5a2481/js/options.js:10:26
```

**Comment by babudro** [comment]
I'm running into the same problem with 57.0.4 under Arch Linux when using an older profile.  When I created a new profile for this version of FF it worked properly.  Perhaps something gets messed up in the transition from legacy API to the new API?

**Comment by tl87** [comment]
Try turning on cookies.. Did it help?

**Comment by ghost** [comment]
Correct! Cookies are whitelist only for me. Adding an exception for `moz-extension://89055bcf-d49f-43f9-bb5b-b0de5e5a2481` via `Firefox Options->General->Exceptions` button makes the popup menu work across random webpages. I had to restart Firefox for the actual settings page to work after I made this change. Thank you for the tip. Warning to others, this *GUID URL* might be different for you so make sure you check which one yours is and use that instead.

**Comment by WUUUGI** [comment]
@tl87 enabling and whitelisting cookies did not help in my case.

@babudro this is the same issue that affects me

I changed the **dom.storage.enabled** config to **true** and SS now displays options on websites, however the SS options page keeps reloading and does not stop to reload.

Setting  **dom.storage.enabled** back to **false** breaks SS completly.
